modules/functional_expansion_tools/src/functions/FunctionSeries.C

```diff
@@ -0,0 +1,269 @@
+  : MutableCoefficientsFunctionInterface(this, this, parameters),
+    _orders(convertOrders(getParam<std::vector<unsigned int>>("orders"))),
+    _physical_bounds(getParam<std::vector<Real>>("physical_bounds")),
+    _series_type_name(getParam<MooseEnum>("series_type")),
+    _x(getParam<MooseEnum>("x")),
+    _y(getParam<MooseEnum>("y")),
+    _z(getParam<MooseEnum>("z")),
+    _disc(getParam<MooseEnum>("disc"))
+{
+  std::vector<MooseEnum> domains;
+  std::vector<MooseEnum> types;
+
+  /*
+   * For Cartesian series, at least one of 'x', 'y', and 'z' must be specified.
+   *
+   * The individual series are always stored in x, y, z order (independent of the order in which
+   * they appear in the input file). Hence, the 'orders' and 'physical_bounds' vectors must always
+   * be specified in x, y, z order.
+   */
+  if (_series_type_name == "Cartesian")
+  {
+    if (isParamValid("x"))
+    {
+      domains.push_back(FBI::_domain_options = "x");
+      types.push_back(_x);
+    }
+    if (isParamValid("y"))
+    {
+      domains.push_back(FBI::_domain_options = "y");
+      types.push_back(_y);
+    }
+    if (isParamValid("z"))
+    {
+      domains.push_back(FBI::_domain_options = "z");
+      types.push_back(_z);
+    }
+    if (types.size() == 0)
+      mooseError("Must specify one of 'x', 'y', or 'z' for 'Cartesian' series!");
+    _series_type = new Cartesian(domains, _orders, types);
+  }
+
+  /*
+   * CylindricalDuo represents a disc-axial expansion, where the disc is described by a single
+   * series, such as Zernike (as opposed to a series individually representing r and a second series
+   * independently representing theta. For CylindricalDuo series, the series are always stored in
+   * the axial, planar order, independent of which order the series appear in the input file.
+   * Therefore, the _orders and _physical_bounds vectors must always appear in axial, planar order.
+   * The first entry in _domains is interpreted as the axial direction, and the following two as the
+   * planar.
+   */
+  if (_series_type_name == "CylindricalDuo")
+  {
+    if (isParamValid("x"))
+    {
+      domains = {
+          FBI::_domain_options = "x", FBI::_domain_options = "y", FBI::_domain_options = "z"};
+      types.push_back(_x);
+    }
+    if (isParamValid("y"))
+    {
+      domains = {
+          FBI::_domain_options = "y", FBI::_domain_options = "x", FBI::_domain_options = "z"};
+      types.push_back(_y);
+    }
+    if (isParamValid("z"))
+    {
+      domains = {
+          FBI::_domain_options = "z", FBI::_domain_options = "x", FBI::_domain_options = "y"};
+      types.push_back(_z);
+    }
+
+    if (types.size() == 0)
+      mooseError("Must specify one of 'x', 'y', or 'z' for 'CylindricalDuo' series!");
+
+    if (types.size() > 1)
+      mooseError("Cannot specify more than one of 'x', 'y', or 'z' for 'CylindricalDuo' series!");
+
+    types.push_back(_disc);
+    _series_type = new CylindricalDuo(domains, _orders, types);
+  }
```

Comment:
Where'd all my virtual destructors go? I must have been too aggressive when removing the empty ones and accidentally removed the ones I actually needed. In practice I always code `new` and `delete` at the same time to prevent this from happening.

I've added it back in. There really is not a good way to do this without raw memory allocation since `FunctionSeries` is working with generic `CompositeSeriesBasisInterface` instances, of which `Cartiesian` and `CylindricalDuo` are the current options. Maybe more to come in the future?

Reply:
I think I misunderstood. Are you referring to using `libmesh_make_unique<T>(...)` or similar instead?

Reply:
`std::shared_ptr<T>`

Reply:
But, if you think no one else should get their dirty hands on your pointers, then `unique_ptr` is good.

Reply:
No touchy

Reply:
> @permcody might bitch about this

You bet I will. I don't want to see the keyword "new" in this new module 😄 .

Reply:
:relaxed: no more `new` anywhere after 3f578e9. Only `std::unique_ptr`.

For completeness the word "new" does show up in a few comments and variable names. :roll_eyes: